packages/twenty-front/src/modules/favorites/components/FavoriteFoldersMultiSelect.tsx

```diff
@@ -0,0 +1,189 @@
+
+import { useFavoriteFoldersScopedStates } from '@/favorites/hooks/useFavoriteFoldersScopedStates';
+import { useMultiFavoriteFolder } from '@/favorites/hooks/useMultiFavoriteFolder';
+import { FavoriteFoldersScopeInternalContext } from '@/favorites/scopes/scope-internal-context/favoritesScopeInternalContext';
+import { ObjectRecord } from '@/object-record/types/ObjectRecord';
+
+import { FAVORITE_FOLDERS_DROPDOWN_ID } from '@/favorites/constants/favoriteFoldersDropdownId';
+import { isFavoriteFolderCreatingState } from '@/favorites/states/isFavoriteFolderCreatingState';
+import { Checkbox } from '@/ui/input/components/Checkbox';
+import { DropdownMenu } from '@/ui/layout/dropdown/components/DropdownMenu';
+import { DropdownMenuSearchInput } from '@/ui/layout/dropdown/components/DropdownMenuSearchInput';
+import { DropdownMenuSeparator } from '@/ui/layout/dropdown/components/DropdownMenuSeparator';
+import { useDropdown } from '@/ui/layout/dropdown/hooks/useDropdown';
+import { MenuItem } from '@/ui/navigation/menu-item/components/MenuItem';
+import { useScopedHotkeys } from '@/ui/utilities/hotkey/hooks/useScopedHotkeys';
+import { useAvailableScopeIdOrThrow } from '@/ui/utilities/recoil-scope/scopes-internal/hooks/useAvailableScopeId';
+import { useTheme } from '@emotion/react';
+import { IconPlus } from 'twenty-ui';
+
+const StyledDropdownContainer = styled.div`
+  position: relative;
+  width: 100%;
+`;
+
+const StyledItemsContainer = styled.div`
+  max-height: 160px;
+  overflow-y: auto;
+`;
+
+const StyledFooter = styled.div`
+  background: ${({ theme }) => theme.background.primary};
+  border-bottom-left-radius: ${({ theme }) => theme.border.radius.md};
+  border-bottom-right-radius: ${({ theme }) => theme.border.radius.md};
+  border-top: 1px solid ${({ theme }) => theme.border.color.light};
+  bottom: 0;
+  position: sticky;
+`;
+const StyledCheckbox = styled(Checkbox)`
+  padding-right: 0;
+`;
+type FavoriteFoldersMultiSelectProps = {
+  onSubmit?: () => void;
+  record?: ObjectRecord;
+  objectNameSingular: string;
+};
+
+const StyledIconPlus = styled(IconPlus)`
+  padding-left: ${({ theme }) => theme.spacing(1)};
+`;
+
+const NO_FOLDER_ID = 'no-folder';
+
+export const FavoriteFoldersMultiSelect = ({
+  onSubmit,
+  record,
+  objectNameSingular,
+}: FavoriteFoldersMultiSelectProps) => {
+  const [isFavoriteFolderCreating, setIsFavoriteFolderCreating] =
+    useRecoilState(isFavoriteFolderCreatingState);
+
+  const theme = useTheme();
+
+  const scopeId = useAvailableScopeIdOrThrow(
+    FavoriteFoldersScopeInternalContext,
+  );
+
+  const { closeDropdown } = useDropdown(FAVORITE_FOLDERS_DROPDOWN_ID);
+
+  const {
+    favoriteFoldersSearchFilterState,
+    favoriteFoldersMultiSelectCheckedState,
+  } = useFavoriteFoldersScopedStates();
+
+  const { getFoldersByIds, toggleFolderSelection } = useMultiFavoriteFolder({
+    record,
+    objectNameSingular,
+  });
+
+  const favoriteFoldersSearchFilter = useRecoilValue(
```

Comment:
instead of useRecoilValue + seSetRecoilState you can use a single useRecoilState